build.gradle

```diff
@@ -67,8 +67,9 @@ dependencies {
     runtime 'org.apache.logging.log4j:log4j-slf4j-impl:2.8.1'
 
     // Database connectors
-    compile 'com.zaxxer:HikariCP:2.6.3'
-    runtime 'org.mariadb.jdbc:mariadb-java-client:2.0.3'
+    compile 'com.zaxxer:HikariCP:3.2.0'
+    runtime 'mysql:mysql-connector-java:8.0.13'
+    runtime 'org.mariadb.jdbc:mariadb-java-client:2.3.0'
```

Comment:
This isn't a criticism of this PR (So please don't make changes without one of the devs chiming in)

But is updates to HikariCP / mariadb required for MySQL 8?

The question comes down to, is the fact that the drivers / pooling is provided by (these particular drivers / pool) an implementation specific thing, that we can change major versions on fine, or is there potential issues with bumping it.

(https://github.com/brettwooldridge/HikariCP/blob/dev/CHANGES#L29) shows that deprecated methods were removed, so there *are* breaking changes if plugins were using HikariCP directly.

At first glance, the mariadb java-client bump seems fine (https://mariadb.com/kb/en/library/mariadb-connector-j-release-notes/).

Reply:
I'm more worried about leaked implementation / Lantern.

Reply:
Honestly, I'm fine with the PR getting merged without Hikari/Maria dependency updates if a proper risk assessment can't be made. I'm unfamiliar with anything not to do with Sponge. Given the impending Java 11 upgrade coming to Minecraft (and thus Forge), the dependencies will need to be updated at some point.

Reply:
I agree that's important to stay with up to date dependencies. But this PR is not a good place to update HikariCP because this is not concerned original issue and because of potential compatibility issues, despite that leaked HikaryCP is completely not sponge problem. If we may maintain compatibility, why won't do that? There is nothing critical in HikariCP changelog that causes to update it asap thus breaking potential 0.000001% of users. It's better to move this update to the next major version or even when minecraft finally would support java 11.